### PR TITLE
Add bpe & unigram tokenizer

### PR DESCRIPTION
**Pull request by joey12300**
### PR types

New features
### PR changes

APIs
### Description

Add bpe and unigram tokenizer, in order to support following models:
1. ERNIE-M, ALBERT, T5, mBART, Big Bird, and XLNet (Supported by unigram).
2. GPT, GPT-2, RoBERTa, BART, and DeBERTa (Supported by bpe).

#### Example
- Prerequisite: Install faster_tokenizer (`pip install faster_tokenizer --upgrade`)
```python
from paddlenlp.transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained('ernie-m-base', use_faster=True)
```

### CODE REVIEW

File: faster_tokenizer/faster_tokenizer/src/utils/lattice.cc [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+#include <random>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "glog/logging.h"
+
+#include "utils/utils.h"
+
+namespace tokenizers {
```
Comment (ZeyuChen) [inline faster_tokenizer/faster_tokenizer/src/utils/lattice.cc]: 我在想我们这个库的namespace是否强绑定下paddlenlp？
namespace paddlenlp {
namespace faster_tokenizer {
}
}

**Review by ZeyuChen — COMMENTED** [review]
整体的库是否突出下paddlenlp的namespace会更有统一品牌的感觉呢？

File: faster_tokenizer/faster_tokenizer/src/utils/lattice.cc (hunk 1 above)
Comment (joey12300) [reply faster_tokenizer/faster_tokenizer/src/utils/lattice.cc]: Done

File: faster_tokenizer/cmake/ByproductsICU.cmake [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
```
Comment (ZeyuChen) [inline faster_tokenizer/cmake/ByproductsICU.cmake]: 去掉无用的空行

Comment (joey12300) [reply faster_tokenizer/cmake/ByproductsICU.cmake]: Done

**Review by ZeyuChen — APPROVED** [review]
LGTM

**Pull request merged**